megatron/core/extensions/transformer_engine.py

```diff
@@ -989,6 +989,12 @@ def __init__(
             self.kept_packed_seq_params.discard("cu_seqlens_q_padded")
             self.kept_packed_seq_params.discard("cu_seqlens_kv_padded")
 
+        # qk-clip is only supported in TE 2.9.0 and later
+        if config.qk_clip and is_te_min_version("2.9.0"):
+            # TE 2.9.0 introduces return_max_logit for qk-clip getting the max attention logits
+            extra_kwargs["return_max_logit"] = True
+            self.current_max_attn_logits = None
```

Comment:
what if `not is_te_min_version` but config.qk_clip? might raise an error

Reply:
Right now it will raise an error if te version is wrong